modules/promotion/src/PromotionStorage.php

```diff
@@ -24,8 +24,27 @@ public function loadValid(OrderTypeInterface $order_type, StoreInterface $store)
     if (empty($result)) {
       return [];
     }
+
+    $usages_query = $this->database->select('commerce_promotion_usage', 'cpu');
```

Comment:
I agree that it's wasteful, but that means we need a new method on the service then. If we keep it this way people won't be able to easily swap the service to use a different storage, which is something that core encourages with its own examples.